pkg/providers/instancetype/suite_test.go

```diff
@@ -1331,6 +1331,80 @@ var _ = Describe("Instance Types", func() {
 			Expect(node.Labels).To(HaveKeyWithValue(v1alpha5.ProvisionerNameLabelKey, provisioner.Name))
 		})
 	})
+	Context("Ephemeral Storage", func() {
+		BeforeEach(func() {
+			nodeTemplate.Spec.AMIFamily = aws.String(v1alpha1.AMIFamilyAL2)
+			nodeTemplate.Spec.BlockDeviceMappings = []*v1alpha1.BlockDeviceMapping{
+				{
+					DeviceName: aws.String("/dev/xvda"),
+					EBS: &v1alpha1.BlockDevice{
+						SnapshotID: aws.String("snap-xxxxxxxx"),
+					},
+				},
+			}
+		})
+		It("should default to EBS defaults when volumeSize is not defined in blockDeviceMappings for custom AMIs", func() {
```

Comment:
optional: Consider using a `DescribeTable` here. There's an example of this style [here](https://github.com/aws/karpenter/blob/2765c329d56451461443e43b04ad7740fb01d30d/test/suites/integration/kubelet_config_test.go#L37)

Reply:
I like the style, but I don't think it helps much since I also have to change the expected device names and ami family strings in the node template. I'm gonna opt to just keep as is